js/ui/file_uploader.js

```diff
@@ -521,9 +574,41 @@ var FileUploader = Editor.inherit({
         }
 
         each(value.slice(this._files.length), (function(_, value) {
-            this._files.push(this._createFile(value));
+            var file = this._createFile(value);
+            this._validateFile(file);
+            this._files.push(file);
         }).bind(this));
     },
+    _validateFile: function(file) {
+        file.isValidFileExtension = this._validateFileExtension(file);
+        file.isValidMinSize = this._validateMinFileSize(file);
+        file.isValidMaxSize = this._validateMaxFileSize(file);
+    },
+    _validateFileExtension: function(file) {
+        var allowedExtensions = this.option("validation.allowedFileExtensions");
+        var fileExtension = file.value.name.substring(file.value.name.lastIndexOf('.'));
+        if(allowedExtensions.length === 0) {
+            return true;
+        }
+        for(var i = 0; i < allowedExtensions.length; i++) {
+            if(fileExtension === allowedExtensions[i]) {
+                return true;
+            }
+        }
+        return false;
+    },
+    _validateMaxFileSize: function(file) {
+        var fileSize = file.value.size;
+        var maxFileSize = this.option("validation.maxFileSize");
+        maxFileSize = maxFileSize === 0 ? fileSize : maxFileSize;
+        return fileSize <= maxFileSize;
```

Comment:
May be `return  maxFileSize > 0 ? fileSize <= maxFileSize : true;`

Reply:
The ternary operator is a condition and depends on the condition, it returns true or false. You proposed a ternary operator, which in the case of true, perform one more condition.
It seems to me that in this case the readability of the code will be worse